docs/source/features/version_control.rst

```diff
@@ -2,19 +2,181 @@
+
+Note that the first "commit" occurred in the third step above is a verb.
+It means the action to turn a draft into a commit.
+
+:numref:`Figure. %s <draft-demonstration>` demonstrates the relations between drafts and commits.
+
+.. _draft-demonstration:
+
+.. figure:: ../images/draft.png
+   :scale: 30 %
+   :align: center
+
+   The relations between a draft and commits.
+
+The following code block creates a draft,
+adds a new segment to the "VersionControlDemo" dataset and does the commit operation.
+
+.. code:: python
+
+   import os
+   from tensorbay.dataset import Segment
+
+   TEST_IMAGES_PATH = "path/to/test_images"
+
+   dataset_client = gas.get_dataset("VersionControlDemo")
+   dataset_client.create_draft("draft-1")
+
+   test_segment = Segment("test")
+
+   for image_name in os.listdir(TEST_IMAGES_PATH):
+       data = Data(os.path.join(TEST_IMAGES_PATH, image_name))
+       test_segment.append(data)
+
+   dataset_client.upload_segment(test_segment, jobs=8)
+   dataset_client.commit("add test segment")
+
+See :ref:`version_control/draft_and_commit:Draft and Commit` for more details about draft.
+
+*****
+ Tag
+*****
+
+For the convenience of marking major commits and switching between different commits,
+TensorBay provides the :ref:`reference/glossary:tag` concept.
+The typical usage of tag is to mark released versions of a dataset.
+
+The tag "v1.0.0" in :numref:`Fig. %s <commit-demonstration>` is added by
+
+.. code:: python
+
+   dataset_client.create_tag("v1.0.0", revision=SECOND_COMMIT_ID)
+
+See :ref:`version_control/tag:Tag` for more details about tag.
+
+********
+ Branch
+********
+
+Sometimes, users may need to create drafts upon an early (not the latest) commit.
+For example, in an algorithm team,
+each team member may do modifications/update based on different versions of the dataset.
+This means a commit list may turn into a commit tree.
+
+For the convenience of maintaining a commit tree, TensorBay provides the :ref:`reference/glossary:branch` concept.
+
+Actually, the commit list (:numref:`Fig. %s <commit-demonstration>`) above is the default branch named "main".
+
+The code block below creates a branch "with-label" based on the :ref:`reference/glossary:revision` "v1.0.0",
+and adds :ref:`classification <reference/label_format/Classification:Classification>` label to the "train" segment.
+
+:numref:`Figure. %s <branch-demonstration>` demonstrates the two branches.
+
+.. _branch-demonstration:
+
+.. figure:: ../images/branch.png
+   :scale: 30 %
    :align: center
 
-   The relations between different versions of a dataset.
+   The relations between branches.
+
+.. code:: python
+
+   TRAIN_IMAGES_PATH = "path/to/train/images"
+
+   dataset_client.create_branch("with-label", revision="v1.0.0")
+   dataset_client.create_draft("draft-2")
+
+   train_segment = Segment("train")
+   train_segment_client = dataset_client.get_segment(train_segment.name)
 
+   for image_name in os.listdir("path/to/train_images/"):
```

Comment:
could we get image_name from dataset["train"] instead of local path?

Reply:
pending